osu.Game/Overlays/Settings/ISettingsItem.cs

```diff
@@ -9,5 +9,11 @@ namespace osu.Game.Overlays.Settings
     public interface ISettingsItem : IDrawable, IDisposable
     {
         event Action SettingChanged;
+
+        /// <summary>
+        /// Apply the default values of a setting item, if the setting item specifies a "classic" default via <see cref="SettingsItem{T}.ApplyClassicDefault"/>.
+        /// </summary>
+        /// <param name="useClassicDefault">Whether to apply the classic value. If <c>false</c>, the standard default is applied.</param>
+        void ApplyClassicDefault(bool useClassicDefault);
```

Comment:
Function name TBD. Though I feel calling this `ApplyDefault` is misleading given the behaviour..

Reply:
Since this is in-code, I think the name `Legacy` explains it best.

Reply:
Actually now that I look at this again, I was misled into thinking this method applies  the classic default without noticing the boolean.

I think `ApplyDefault(bool legacyDefault = false)` (or classic) makes sense.

Reply:
Could also consider splitting into two methods, or declaring an enum type (`Standard`/`Classic` defaults) rather than passing a boolean. Maybe too much of a muchness though.

Reply:
I was going to use the above recommendation, but just want to be sure that everyone understands that this method will **not apply anything** if the parameter is set to `true` and the setting doesn't have a classic default. I'm not sure how well a method named `ApplyDefault` communicates this.

Reply:
But is that going to be the case in the behaviour screen? I thought it would call `ApplyDefault` on just those that have a classic default, ignoring the others.

At which point I would think it should throw on the "`true` but no classic" case.

Reply:
So you mean expose the `HasClassicDefault` as well?

Reply:
That's one way, but another would be checking if the setting is present in `BehaviourScreen` during `applyClassic`/`applyStandard` (assuming that the only settings that are present in the screen have a classic default).

I think exposing `HasClassicDefault` sounds better, for the usage to guard against the exception by checking `HasClassicDefault` first.

Reply:
Agree with exposing `HasClassicDefault` and using that at the point where `ApplyClassicDefault()` is currently being called (in some loop over all settings that have a classic default, I imagine). `ApplyClassicDefault()` method is just not a well-defined method right now, it has too many conditional thingies to be well-defined and the naming woes are just a symptom of that.

Reply:
Sounds good. Should the `ApplyClassicDefault` throw if called incorrectly (when `HasClassicDefault` is `false`)?

Reply:
Not sure I understand the question. The way I'd see it is that `ApplyClassicDefault()` would cease to exist entirely and `HasClassicDefault`/`ClassicDefault` would be exposed as data to be used externally and directly by the behaviour screen.

Reply:
This wouldn't work due to generics (the reason I structured it this way in the first place). ie. we are dealing with `ISettingsItem`s because there are various `<T>` usages of them that we can't match/operate over.

Reply:
Fair enough. Throwing seems reasonable in that case. I presume the only thing that `ApplyClassicDefault` would just check that a classic default exists (throwing if it doesn't) and then just apply it unconditionally, right?

Reply:
It will still need the `bool` as well, unless we expose a second method.

Reply:
Second method seems better. Otherwise it's pretty much back to square one for me with the boolean still there.

Reply:
to confirm, does this look about what you'd expect from the exposed methods?

```csharp
    public interface ISettingsItem : IDrawable, IDisposable
    {
        event Action SettingChanged;

        /// <summary>
        /// Whether this setting has a classic default (ie. a different default which better aligns with osu-stable expectations).
        /// </summary>
        bool HasClassicDefault { get; }

        /// <summary>
        /// Apply the classic default value of the associated setting. Will throw if <see cref="HasClassicDefault"/> is <c>false</c>.
        /// </summary>
        void ApplyClassicDefault();

        /// <summary>
        /// Apply the default value of the associated setting.
        /// </summary>
        void ApplyDefault();
    }

```

Reply:
Yeah, that's pretty much the shape I'd expect to see.